specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -3332,6 +3332,475 @@
+          },
+          "DataFlows_Update": {
+            "$ref": "./examples/DataFlows_Update.json"
+          }
+        },
+        "description": "Creates or updates a data flow.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/dataFlowName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "ETag of the data flow entity. Should only be specified for update, for which it should match existing entity or can be * for unconditional update."
+          },
+          {
+            "name": "dataFlow",
+            "description": "Data flow resource definition.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DataFlowResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/DataFlowResource"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Data Factory service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "dataFlows"
+        ],
+        "operationId": "DataFlows_Get",
+        "x-ms-examples": {
+          "DataFlows_Get": {
+            "$ref": "./examples/DataFlows_Get.json"
+          }
+        },
+        "description": "Gets a data flow.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/dataFlowName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "name": "If-None-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "ETag of the data flow entity. Should only be specified for get. If the ETag matches the existing entity tag, or if * was provided, then no content will be returned."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/DataFlowResource"
+            }
+          },
+          "304": {
+            "description": "Not modified."
+          },
```

Comment:
As per the RPC, GETs should always return 200 (unless there's a failure)

Reply:
Got it, it's actually the same pattern as all other sub resources we have. I will remove it here and change our RP at the same time

Reply:
Sounds good.

---
In reply to: [325400189](https://github.com/Azure/azure-rest-api-specs/pull/7088#discussion_r325400189) [](ancestors = 325400189)

Reply:
@majastrz 304 is for "If-None-Match" header.

{
--
  | "name": "If-None-Match",
  | "in": "header",
  | "required": false,
  | "type": "string",
  | "description": "ETag of the dataset entity. Should only be specified for get. If the ETag matches the existing entity tag, or if * was provided, then no content will be returned."
  | }